source/Octopus.Tentacle.Contracts/IAuthContext.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+
+namespace Octopus.Tentacle.Contracts
+{
+    public interface IAuthContext
```

Comment:
Just had the thought, is `IAuthContext` too generic? Would `IScriptAuthContext` be a bit clearer (also thinking about how it manifests in Server)

Reply:
I'm not sure I see the value of having this interface for extension with only one implementation. Are we already expecting more different kinds of auth contexts for Kubernates commands? Is it hard to add this extension point later if we need it?